tools/generate_snapshot.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env python
+import gzip
+import os
+
+import click
+import simplejson
+from eth_utils import to_canonical_address, to_hex
+from requests.exceptions import ConnectionError
+from web3 import HTTPProvider, Web3
+
+from raiden.blockchain.events import fetch_all_events_for_a_deployment
+from raiden.network.rpc.client import JSONRPCClient
+from raiden.settings import DEFAULT_NUMBER_OF_BLOCK_CONFIRMATIONS, RAIDEN_CONTRACT_VERSION
+from raiden.tests.utils import factories
+from raiden.ui.app import rpc_normalized_endpoint
+from raiden.ui.checks import (
+    check_ethereum_client_is_supported,
+    check_ethereum_network_id,
+    check_synced,
+)
+from raiden.ui.cli import ETH_NETWORKID_OPTION, ETH_RPC_CONFIG_OPTION
+from raiden.utils.cli import NetworkChoiceType, group, option
+from raiden.utils.formatting import to_checksum_address
+from raiden.utils.typing import (
+    BlockNumber,
+    Dict,
+    SecretRegistryAddress,
+    TokenNetworkRegistryAddress,
+)
+from raiden_contracts.constants import (
+    CONTRACT_SECRET_REGISTRY,
+    CONTRACT_TOKEN_NETWORK_REGISTRY,
+    EVENT_TOKEN_NETWORK_CREATED,
+    ID_TO_CHAINNAME,
+    ChannelEvent,
+)
+from raiden_contracts.contract_manager import (
+    ContractManager,
+    contracts_precompiled_path,
+    get_contracts_deployment_info,
+)
+
+
+def format_event_for_serialization(data: Dict):
```

Comment:
The workflow here is unnecessary convoluted but I didn't want to refactor this now:

- web3 fetches the data and does it's own decoding
- then we decode the events using the contracts abi
- then we proceed to do our sandwich encoding, converting hex to bytes
- all of the above is done for internal usage, since I reused the code, now I have to undo the sandwich encoding, otherwise the data cannot be json encoded